tests/Unit/Infrastructure/JsonStringReaderTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\Infrastructure;
+
+use FileFetcher\InMemoryFileFetcher;
+use FileFetcher\ThrowingFileFetcher;
+use WMDE\Fundraising\Frontend\Infrastructure\JsonStringReader;
+
+class JsonStringReaderTest extends \PHPUnit\Framework\TestCase {
+
+	public function testGivenFailingFileFetcher_exceptionIsThrown() {
+		$this->expectException( \RuntimeException::class );
+		( new JsonStringReader(
+			'wrong.file',
+			'/test/directory/',
+			new ThrowingFileFetcher()
+		) )->validateJson();
+	}
+
+	public function testGivenEmptyJsonFile_exceptionIsThrown() {
+		$this->expectException( \RuntimeException::class );
+		( new JsonStringReader(
+			'empty.json',
+			'/test/directory/',
+			$this->getFileFetcherWithContent( [ '/test/directory/empty.json' => '' ] )
+		) )->validateJson();
+	}
+
+	private function getFileFetcherWithContent( array $content ): InMemoryFileFetcher {
+		return new InMemoryFileFetcher( $content );
+	}
+
+	public function testGivenInvalidJsonFile_exceptionIsThrown() {
+		$this->expectException( \RuntimeException::class );
+		( new JsonStringReader(
+			'invalid.json',
+			'/test/directory/',
+			$this->getFileFetcherWithContent( [ '/test/directory/invalid.json' => '{"test": "is" "broken": "now "}' ] )
```

Comment:
This faulty JSON is very subtle. Maybe you can make the error a bit more obvious (e.g. leaving out quotes and last brace)